tempesta_fw/sock.c

```diff
@@ -1162,17 +1256,122 @@ ss_tx_action(void)
 		default:
 			BUG();
 		}
+dead_sock:
 		sock_put(sk); /* paired with tfw_wq_push() */
 	}
 }
 
+#define HANDLE_TOO_LONG_WAIT(t0, job)					\
+do {									\
+	if (t0 + HZ * 5 < jiffies) {					\
+		SS_WARN("cpu %d: pending " job " for 5s (__ss_active=%#lx)\n",\
+			cpu, atomic64_read(active));			\
+		return;	/* give up - everything should be finished now */\
```

Comment:
Yes, the problem is with #668: sometimes we just can't accurately account sockets. I added comment to the issue https://github.com/tempesta-tech/tempesta/issues/668#issuecomment-276735774 and adjust the comment in the macro.